web/packages/shared/components/Controls/ViewModeSwitch.story.tsx

```diff
@@ -0,0 +1,61 @@
+/**
+ * Teleport
+ * Copyright (C) 2024 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useState } from 'react';
+import { Flex } from 'design';
+
+import { ViewMode } from 'gen-proto-ts/teleport/userpreferences/v1/unified_resource_preferences_pb';
+
+import { ViewModeSwitch } from './ViewModeSwitch';
+
+import type { Meta, StoryFn, StoryObj } from '@storybook/react';
+
+export default {
+  title: 'Shared/Controls/ViewModeSwitch',
+  component: ViewModeSwitch,
+  argTypes: {
+    currentViewMode: {
+      control: { type: 'radio', options: [ViewMode.CARD, ViewMode.LIST] },
+      description: 'Current view mode',
+      table: { defaultValue: { summary: ViewMode.CARD.toString() } },
+    },
+    setCurrentViewMode: {
+      control: false,
+      description: 'Callback to set current view mode',
+      table: { type: { summary: '(newViewMode: ViewMode) => void' } },
+    },
+  },
+  args: { currentViewMode: ViewMode.CARD },
+  parameters: { controls: { expanded: true, exclude: ['userContext'] } },
+} satisfies Meta<typeof ViewModeSwitch>;
+
+const Default = {
+  render: (({ currentViewMode }) => {
+    const [viewMode, setViewMode] = useState(currentViewMode);
+    return (
+      <Flex alignItems="center" minHeight="100px">
+        <ViewModeSwitch
+          currentViewMode={viewMode}
+          setCurrentViewMode={setViewMode}
+        />
+      </Flex>
+    );
+  }) satisfies StoryFn<typeof ViewModeSwitch>,
+} satisfies StoryObj<typeof ViewModeSwitch>;
```

Comment:
Thoughts on defining the type for `Default` over using `satisfies`? IIRC providing just the type (or `satisfies`) for `Default` should be enough and TypeScript should be able to infer the rest.

```typescript
const Default: StoryObj<typeof ViewModeSwitch> = {
  render: (({ currentViewMode }) => {
    const [viewMode, setViewMode] = useState(currentViewMode);
    return (
      <Flex alignItems="center" minHeight="100px">
        <ViewModeSwitch
          currentViewMode={viewMode}
          setCurrentViewMode={setViewMode}
        />
      </Flex>
    );
  }),
};
```

Reply:
`export default` is a little bit more difficult as we'd be forced to do `const meta: Meta<foo>` and then `export default meta`.

Reply:
hm, I can try that way, but I found I needed at least the `Satisfies` for `render`, since otherwise TS didn't recognize it as a React fn and gave a spurious warning for using `useState`

Reply:
I haven't looked at how it works exactly, but I think it's not TS that does that. It seems to me that it might be that once you add `satisfies`, the `react-hooks` ESLint plugin bails out and it stops complaining. You can verify that by removing all types from `Default` and replacing that `satifies` with `satisfies any`.

The warning message calls out that the React component names must start with an uppercase letter. The warning indeed goes away when you change `render` to `Render`, but ofc then that's not valid Storybook setup.

Reply:
* https://github.com/storybookjs/storybook/issues/25294
* https://github.com/storybookjs/storybook/issues/21115

I guess the mentioned workaround means using CSF2. Personally I'd remove `satisfies` and lived with the warning for now.